hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -1245,8 +1245,102 @@ def test_not_equal(x, y):
         Without the ability to reorder x and y this could fail either with
         ``x=""``, ``y="0"``, or the other way around. With reordering it will
         reliably fail with ``x=""``, ``y="0"``.
+
+        This pass ignores block examples. This is because there are a lot
+        of single block examples (many of which may not be well labelled),
+        and such reorderings are much less likely to work. We could add
+        an emergency shrink pass to patch up any possible block reorderings,
+        but so far there doesn't seem to be a compelling use case for doing
+        so.
         """
-        self.example_wise_shrink(Ordering, key=sort_key)
+
+        # Rather than index order we want to iterate over examples in order
+        # of increasing depth, with index order among the examples at a given
+        # depth. This retains the property that our iteration is stable in the
+        # sense that if we replace the current example, its replacement lives
+        # in the same index in the new list (which we have to recalculate).
+        # However, it ensures that we start by trying larger examples,
+        # prioritising big swaps. This prevents us getting bogged down too
+        # early and gives us more opportunities to luck into a successful
+        # reduction of the size of the test case.
+        def calc_examples():
+            return sorted(
+                self.shrink_target.examples, key=lambda ex: (ex.depth, ex.index)
+            )
+
+        examples = calc_examples()
+
+        # Note our iteration order is that we repeatedly try to make the
+        # example at index i smaller.
+        i = 0
+        while i < len(examples):
+            succeeded = False
+            ex1 = examples[i]
+            buf = self.buffer
+            b1 = buf[ex1.start : ex1.end]
+
+            if ex1.children:
+                candidates = [
```

Comment:
This could use an up-front comment explaining that we're compiling a list of examples that would be good candidates for swapping with `ex1`, to see if that results in a shrink.